SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -1793,3 +1796,107 @@ void USpatialNetDriver::DelayedSendDeleteEntityRequest(Worker_EntityId EntityId,
 		Sender->SendDeleteEntityRequest(EntityId);
 	}, Delay, false);
 }
+
+void USpatialNetDriver::HandleStartupOpQueueing(const TArray<Worker_OpList*>& InOpLists)
+{
+	if (InOpLists.Num() == 0)
+	{
+		return;
+	}
+
+	QueuedStartupOpLists.Append(InOpLists);
+	bQueueOpsUntilReady = FindAndDispatchStartupOps(InOpLists);
+
+	// We've found and dispatched all ops we need for startup, trigger BeginPlay()
+	// on the GSM and process the queued ops.  Note that FindAndDispatchStartupOps()
+	// will have notified the Dispatcher to skip the startup ops that we've
+	// processed already.
+	if (!bQueueOpsUntilReady)
+	{
+		GlobalStateManager->TriggerBeginPlay();
+
+		for (Worker_OpList* OpList : QueuedStartupOpLists)
+		{
+			Dispatcher->ProcessOps(OpList);
+			Worker_OpList_Destroy(OpList);
+		}
+
+		// Sanity check that the dispatcher encountered, skipped, and removed
+		// all Ops we asked it to skip
+		check(Dispatcher->GetNumOpsToSkip() == 0);
+
+		QueuedStartupOpLists.Empty();
+	}
+}
+
+bool USpatialNetDriver::FindAndDispatchStartupOps(const TArray<Worker_OpList*>& InOpLists)
+{
+	for (Worker_OpList* OpList : InOpLists)
```

Comment:
What do you think if `FindFirstOpOfType` takes in a `TArray<Worker_OpList*>` rather than an OpList? Then you could pass in the whole list and take out the for loop here.

Reply:
On second thought, that's probably less performant. Probably doesn't matter.

Reply:
Yes, I actually had it like that originally, then changed it to assist with an approach that I later abandoned (cutting the oplists).  If you'd prefer I can change it back